packages/edit-post/src/components/start-page-options/index.js

```diff
@@ -1,115 +1,44 @@
-	// the current post type is part of the postTypes declared.
-	const { blockPatternsWithPostContentBlockType, postType } = useSelect(
-		( select ) => {
-			const { getPatternsByBlockTypes } = select( blockEditorStore );
-			const { getCurrentPostType } = select( editorStore );
-			return {
-				blockPatternsWithPostContentBlockType:
-					getPatternsByBlockTypes( 'core/post-content' ),
-				postType: getCurrentPostType(),
-			};
-		},
-		[]
-	);
-
-	return useMemo( () => {
-		// filter patterns without postTypes declared if the current postType is page
-		// or patterns that declare the current postType in its post type array.
-		return blockPatternsWithPostContentBlockType.filter( ( pattern ) => {
-			return (
-				( postType === 'page' && ! pattern.postTypes ) ||
-				( Array.isArray( pattern.postTypes ) &&
-					pattern.postTypes.includes( postType ) )
-			);
-		} );
-	}, [ postType, blockPatternsWithPostContentBlockType ] );
-}
-
-function PatternSelection( { blockPatterns, onChoosePattern } ) {
-	const shownBlockPatterns = useAsyncList( blockPatterns );
-	const { resetEditorBlocks } = useDispatch( editorStore );
-	return (
-		<BlockPatternsList
-			blockPatterns={ blockPatterns }
-			shownPatterns={ shownBlockPatterns }
-			onClickPattern={ ( _pattern, blocks ) => {
-				resetEditorBlocks( blocks );
-				onChoosePattern();
-			} }
-		/>
-	);
-}
-
-function StartPageOptionsModal() {
-	const [ modalState, setModalState ] = useState( 'initial' );
-	const startPatterns = useStartPatterns();
-	const hasStartPattern = startPatterns.length > 0;
-	const shouldOpenModal = hasStartPattern && modalState === 'initial';
-
-	useEffect( () => {
-		if ( shouldOpenModal ) {
-			setModalState( 'open' );
-		}
-	}, [ shouldOpenModal ] );
-
-	if ( modalState !== 'open' ) {
-		return null;
-	}
-
-	return (
-		<Modal
-			className="edit-post-start-page-options__modal"
-			title={ __( 'Choose a pattern' ) }
-			isFullScreen
-			onRequestClose={ () => setModalState( 'closed' ) }
-		>
-			<div className="edit-post-start-page-options__modal-content">
-				<PatternSelection
-					blockPatterns={ startPatterns }
-					onChoosePattern={ () => setModalState( 'closed' ) }
-				/>
-			</div>
-		</Modal>
-	);
-}
-
 export default function StartPageOptions() {
-	const shouldEnableModal = useSelect( ( select ) => {
-		const { isCleanNewPost } = select( editorStore );
+	const [ isClosed, setIsClosed ] = useState( false );
+	const { resetEditorBlocks } = useDispatch( editorStore );
+	const { shouldOpenModal, postType } = useSelect( ( select ) => {
+		const { isCleanNewPost, getCurrentPostType } = select( editorStore );
 		const { isEditingTemplate, isFeatureActive } = select( editPostStore );
 
-		return (
-			! isEditingTemplate() &&
-			! isFeatureActive( 'welcomeGuide' ) &&
-			isCleanNewPost()
-		);
+		return {
+			shouldOpenModal:
+				! isEditingTemplate() &&
+				! isFeatureActive( 'welcomeGuide' ) &&
+				isCleanNewPost(),
```

Comment:
I don't think it's possible to replicate `isCleanNewPost` in the site editor because it checks for pages in the `auto-draft` status (never been saved). The site editor creates new pages as `draft`. The main drawback is that if the page has no content when saved and you reload, the modal shows again.

Reply:
The modal also showed again for me if the post did have content.

Reply:
I had a play around with this and the only way I could get it to not load the modal when refreshing a draft page with content was by adding something like:

```js
			const pageRecord = getEntityRecord(
				'postType',
				'page',
				context?.postId
			);
			const hasContent = pageRecord?.content?.raw !== '';

			if ( hasContent || ! isEditingPage || isWelcomeGuideOpen ) {
				return { shouldOpenModal: false };
			}
```
I don't think it is a big issue for the modal to load again on a page refresh of a still empty page, but it might be worth stopping it for pages with content. But maybe I am missing something if you are only seeing it on empty page reloads?

Reply:
> The main drawback is that if the page has no content when saved and you reload, the modal shows again.

We are protected against this, because a completely empty page cannot be saved. It needs to have a non-empty title, excerpt or content. See the `isEditedPostSaveable` selector.